### ISSUE TITLE
SoapFault results in error

### ISSUE THREAD

**Issue by nathansamson** [savonrb/savon#626]
Getting a specific Soap Response which raises SoapFault results in "undefined method `each' for nil:NilClass" when converting that exception to a string.

~~~
SOAP response (status 200)
<?xml version="1.0" encoding="UTF-8"?>
<soap:Envelope xmlns:soap="http://schemas.xmlsoap.org/soap/envelope/">
  <soap:Body>
    <ServiceFault xmlns="http://redacted">
      <faultcode>5001</faultcode>
      <faultstring>REDACTED</faultstring>
      <faultactor>REDACTED</faultactor>
      <detail>REDACTED</detail>
    </ServiceFault>
  </soap:Body>
</soap:Envelope>
~~~

Which results in following stacktrace
nori (2.4.0) lib/nori.rb:71:in `find_value'
nori (2.4.0) lib/nori.rb:37:in `find'
savon (2.5.1) lib/savon/soap_fault.rb:34:in `message_by_version'
savon (2.5.1) lib/savon/soap_fault.rb:23:in `to_s'

**Comment by pichot** [comment]
Any progress on this? I've encountered the same error.

@nathansamson what solution did you come up? Even if it's not PRable, would appreciate seeing it. Thanks!

**Comment by tjarratt** [comment]
This should have been addressed by merging #630.

@nathansamson and @pichot is there more we can do to improve the handling of ServiceFaults?

For what it's worth, I've just published Savon v2.9.0 that includes the changes @nathansamson submitted for pull request #630.

**Comment by tjarratt** [comment]
Closing since this should have been fixed by the previous merge.
